include/libc/assert.h

```diff
@@ -0,0 +1,41 @@
+#ifndef ASSERT_H
+#define ASSERT_H
+
+#if !defined(__GNUC__) && !defined(__attribute__)
+#define __attribute__(x)
+#endif
+
+// Runtime assertions
+
+__attribute__((noreturn)) void __assert(const char* assertion, const char* file, int line);
+
+#ifndef NDEBUG
+# ifndef NON_MATCHING
+#  define ASSERT(cond, msg, file, line) ((cond) ? ((void)0) : __assert(msg, file, line))
+# else
+// macro for matching
+#  define ASSERT(cond, msg, file, line) ((cond) ? ((void)0) : __assert(#cond, __FILE__, __LINE__))
+// standard assert macro
+#  define assert(cond) ASSERT(cond, ., ., .)
+# endif
+#else
+# define ASSERT(cond, msg, file, line) ((void)0)
+# define assert(cond) ((void)0)
+#endif
+
+// Static/compile-time assertions
+
+#if defined(__GNUC__) || (__STDC_VERSION__ >= 201112L)
```

Comment:
Wouldn't just
```suggestion
#if (__STDC_VERSION__ >= 201112L)
```
be enough / make more sense? (unless gcc did have the macro way before C11)